server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -342,134 +342,143 @@ describe("Routerlicious", () => {
-										count == 0 ? signalContent1 : signalContent2,
-										signal.content,
-										"Signal content mismatch",
-									);
-									assert.equal(
-										firstConnectMessage.clientId,
-										signal.clientId,
-										"Client ID mismatch",
-									);
-									count++;
-
-									if (count == 2) {
+					describe("v1 signals", () => {
+						let firstSocket: LocalWebSocket;
+						let firstConnectMessage: IConnected;
+						let secondSocket: LocalWebSocket;
+
+						beforeEach(async () => {
+							firstSocket = webSocketServer.createConnection();
+							firstConnectMessage = await connectToServer(
+								testId,
+								testTenantId,
+								testSecret,
+								firstSocket,
+							);
+							secondSocket = webSocketServer.createConnection();
+							await connectToServer(testId, testTenantId, testSecret, secondSocket);
+						});
+						it("Single Signal", async () => {
+							const signalContent = "TestSignal";
+							const promises = [
+								new Promise<void>((resolve) => {
+									firstSocket.on("signal", (signal: ISignalMessage) => {
+										assert.equal(
+											signalContent,
+											signal.content,
+											"Signal content mismatch",
+										);
+										assert.equal(
+											firstConnectMessage.clientId,
+											signal.clientId,
+											"Client ID mismatch",
+										);
 										resolve();
-									}
-								});
-							}),
-							new Promise<void>((resolve) => {
-								let count = 0;
-
-								secondSocket.on("signal", (signal: ISignalMessage) => {
-									assert.equal(
-										count == 0 ? signalContent1 : signalContent2,
-										signal.content,
-										"Signal content mismatch",
-									);
-									assert.equal(
-										firstConnectMessage.clientId,
-										signal.clientId,
-										"Client ID mismatch",
-									);
-									count++;
-
-									if (count == 2) {
+									});
+								}),
+								new Promise<void>((resolve) => {
+									secondSocket.on("signal", (signal: ISignalMessage) => {
+										assert.equal(
+											signalContent,
+											signal.content,
+											"Signal content mismatch",
+										);
+										assert.equal(
+											firstConnectMessage.clientId,
+											signal.clientId,
+											"Client ID mismatch",
+										);
 										resolve();
-									}
-								});
-							}),
-						];
+									});
+								}),
+							];
+							firstSocket.send("submitSignal", firstConnectMessage.clientId, [
+								signalContent,
+							]);
+							await Promise.all(promises);
+						});
 
-						firstSocket.send("submitSignal", firstConnectMessage.clientId, [
-							signalContent1,
-						]);
-						firstSocket.send("submitSignal", firstConnectMessage.clientId, [
-							signalContent2,
-						]);
-						await Promise.all(promises);
+						it("Multiple Signals", async () => {
+							const signalContent1 = "TestSignal1";
+							const signalContent2 = "TestSignal2";
+
+							const promises = [
```

Comment:
Can you please make this more readable? Examples: name of the variable (or const) could better reflect what it holds, arrays elements could be instantiated outside of the array itself etc. Get rid of duplications through helper functions.